File: C/LinearSearch.c
#include <stdio.h>
#include "LinearSearch.h"
/**
 * Performs linear search to find target element in array
 * @param arr Array to search in
 * @param size Size of the array
 * @param target Element to find
 * @return Index of target if found, -1 otherwise
 */
int linear_search(int arr[], int size, int target) {
  for (int i = 0; i < size; i++) {
    if (arr[i] == target) {
      return i;
    }
  }
  return -1;
}

// int main(void) {
//   int arr[] = {1, 2, 3, 4, 5};
//   int size = sizeof(arr) / sizeof(arr[0]);
//   int target = 3;

//   int idx = linear_search(arr, size, target);
//   if (idx == -1) {
//     printf("Not found");
//   } else {
//     printf("Element %d found at index %d", target, idx);
//   }

//   return 0;
// }

File: python/LinearSearch.py
from typing import Sequence


def linear_search(array: Sequence, target):
    """
    Perform a linear search on the given sequence to find the target element.

    Parameters:
    array (Sequence): The sequence to search through.
    target: The element to search for.

    Returns:
    int: The index of the target element if found, otherwise -1.
    """
    for idx, element in enumerate(array):
        if element == target:
            return idx
    return -1


if __name__ == "__main__":
    arr = [1, 2, 3, 4, 5]
    target = 3
    found_idx = linear_search(arr, target)  # 2
    if found_idx != -1:
        print(f"Element found at index {found_idx} in the array")
    else:
        print(f"Element {target} not found in the array")

    # Test cases are in the test directory.
